src/vs/workbench/browser/parts/compositeBar.ts

```diff
@@ -29,12 +30,18 @@ export interface ICompositeBarItem {
 	visible: boolean;
 }
 
+export interface ICompositeBarDndHandler {
```

Comment:
While this `interface` makes sense, I would look for the Tree interface for inspiration. It might use a good aproach, and it might be cleanier to use something like that. Also for consistency. Check it out as inspiration [here](https://github.com/microsoft/vscode/blob/d793ae615eb7fc61555bd1ea9dff12203a4992f6/src/vs/base/browser/ui/list/list.ts#L104)